File: src/dataset/c/perfect-numbers_exercism-viettronics868/gemini/GeminiFlash_perfect_numbers_v3.c
#include "perfect_numbers.h"
#include <stdint.h>

int classify_number(int number) {
    if (number <= 0) {
        return ERROR;
    }

    int sum = 1; 
    for (int i = 2; i * i <= number; ++i) {
        if (number % i == 0) {
            sum += i;
            if (i * i != number) {
                sum += number / i;
            }
        }
    }

    if (sum == number) {
        return PERFECT_NUMBER;
    } else if (sum > number) {
        return ABUNDANT_NUMBER;
    } else {
        return DEFICIENT_NUMBER;
    }
}

File: src/dataset/typescript/perfect-numbers_exercism-samajammin/gemini/GeminiFlash_perfect-numbers_v3.ts
export default class PerfectNumbers {
  public static readonly PERFECT = 'perfect';
  public static readonly ABUNDANT = 'abundant';
  public static readonly DEFICIENT = 'deficient';

  public static getSumOfFactors(int: number): number {
    let sum = 1;
    const limit = Math.sqrt(int);

    for (let i = 2; i <= limit; i++) {
      if (int % i === 0) {
        sum += i;
        if (i * i !== int) {
          sum += int / i;
        }
      }
    }
    return sum;
  }

  public static classify(int: number): string {
    if (int < 1) {
      throw new Error('Classification is only possible for natural numbers.');
    }

    const sum = PerfectNumbers.getSumOfFactors(int);

    if (sum === int) {
      return PerfectNumbers.PERFECT;
    } else if (sum > int) {
      return PerfectNumbers.ABUNDANT;
    } else {
      return PerfectNumbers.DEFICIENT;
    }
  }
}